internal/querycoordv2/balance/rowcount_based_balancer.go

```diff
@@ -0,0 +1,191 @@
+package balance
+
+import (
+	"sort"
+
+	"github.com/milvus-io/milvus/internal/querycoordv2/meta"
+	"github.com/milvus-io/milvus/internal/querycoordv2/session"
+	"github.com/milvus-io/milvus/internal/querycoordv2/task"
+)
+
+type RowCountBasedBalancer struct {
+	RoundRobinBalancer
+	nodeManager *session.NodeManager
+	dist        *meta.DistributionManager
+	meta        *meta.Meta
+}
+
+func (b *RowCountBasedBalancer) AssignSegment(segments []*meta.Segment, nodes []int64) []SegmentAssignPlan {
+	if len(nodes) == 0 {
+		return nil
+	}
+	nodeItems := b.convertToNodeItems(nodes)
+	queue := newPriorityQueue()
+	for _, item := range nodeItems {
+		queue.push(item)
+	}
+
+	sort.Slice(segments, func(i, j int) bool {
+		return segments[i].GetNumOfRows() > segments[j].GetNumOfRows()
+	})
+
+	plans := make([]SegmentAssignPlan, 0, len(segments))
+	for _, s := range segments {
+		// pick the node with the least row count and allocate to it.
```

Comment:
Maybe there needs to add a quit condition when the row count is nearly equal for all nodes, or it will be skewed after traversing completely.

Reply:
> Maybe there needs to add a quit condition when the row count is nearly equal for all nodes, or it will be skewed after traversing completely.

Nodes are sorted in a priority queue. We pick the node with least row count every time. And this interface is to allocate segments, so we should quit after all segments are allocated.